include/deal.II/lac/la_parallel_block_vector.h

```diff
@@ -0,0 +1,599 @@
+
+      /**
+       * Scaling and simple vector addition, i.e.  <tt>*this =
+       * s*(*this)+V</tt>.
+       */
+      void sadd (const Number               s,
+                 const BlockVector<Number> &V);
+
+      /**
+       * Assignment <tt>*this = a*u + b*v</tt>.
+       *
+       * This function is deprecated.
+       */
+      void equ (const Number a, const BlockVector<Number> &u,
+                const Number b, const BlockVector<Number> &v) DEAL_II_DEPRECATED;
+
+      /**
+       * Scaling and multiple addition.
+       *
+       * This function is deprecated.
+       */
+      void sadd (const Number               s,
+                 const Number               a,
+                 const BlockVector<Number> &V,
+                 const Number               b,
+                 const BlockVector<Number> &W) DEAL_II_DEPRECATED;
+
+      /**
+       * Return whether the vector contains only elements with value zero.
+       * This function is mainly for internal consistency checks and should
+       * seldom be used when not in debug mode since it uses quite some time.
+       */
+      bool all_zero () const;
+
+      /**
+       * Computes the mean value of all the entries in the vector.
+       */
+      Number mean_value () const;
+
+      /**
+       * $l_p$-norm of the vector. The pth root of the sum of the pth powers
+       * of the absolute values of the elements.
+       */
+      real_type lp_norm (const real_type p) const;
+
+      /**
+       * Swap the contents of this vector and the other vector <tt>v</tt>. One
+       * could do this operation with a temporary variable and copying over
+       * the data elements, but this function is significantly more efficient
+       * since it only swaps the pointers to the data of the two vectors and
+       * therefore does not need to allocate temporary storage and move data
+       * around.
+       *
+       * Limitation: right now this function only works if both vectors have
+       * the same number of blocks. If needed, the numbers of blocks should be
+       * exchanged, too.
+       *
+       * This function is analog to the the swap() function of all C++
+       * standard containers. Also, there is a global function swap(u,v) that
+       * simply calls <tt>u.swap(v)</tt>, again in analogy to standard
+       * functions.
+       */
+      void swap (BlockVector<Number> &v);
+      //@}
+
+      /**
+       * @name 2: Implementation of VectorSpaceVector
+       */
+      //@{
+
+      /**
+       * Multiply the entire vector by a fixed factor.
+       */
+      virtual BlockVector<Number> &operator*= (const Number factor);
+
+      /**
+       * Divide the entire vector by a fixed factor.
+       */
+      virtual BlockVector<Number> &operator/= (const Number factor);
+
+      /**
+       * Add the vector @p V to the present one.
+       */
+      virtual BlockVector<Number> &operator+= (const VectorSpaceVector<Number> &V);
```

Comment:
Does this really make sense? I am wondering if this class should really derive from ```VectorSpaceVector```.